lib/image-stream/archive.js

```diff
@@ -195,6 +195,7 @@ exports.extractImage = (archive, hooks) => {
     }).then((results) => {
       results.metadata.stream = results.imageStream;
       results.metadata.transform = new PassThroughStream();
+      results.metadata.path = archive;
```

Comment:
This particular test is covering that: https://github.com/resin-io/etcher/blob/f242443d29864346ef9860879ea447bded283220/tests/image-stream/zip.spec.js#L74